src/Common/MemoryBufferWriter.cs

```diff
@@ -5,11 +5,13 @@
 using System.Buffers;
 using System.Collections.Generic;
 using System.IO;
+using System.Runtime.CompilerServices;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace Microsoft.AspNetCore.Internal
 {
-    internal sealed class MemoryBufferWriter : IBufferWriter<byte>
+    internal sealed class MemoryBufferWriter : Stream, IBufferWriter<byte>
```

Comment:
The name doesn't fit anymore. How about `MemoryWriter`?

Reply:
I was waiting for you to request a rename.

Reply:
Why doesn't the name fit? MemoryStreamWriter? Should it encapsulate everything it's implementing? 😄

Reply:
Also renaming a shared source file is a bit of a chore so I'd like to avoid it as part of this PR (plus  the name is fine).

Reply:
`MemoryStreamBufferWriter`?

Reply:
Should I added pooled in there too (I'm trolling BTW).